scripts/build/termux_step_start_build.sh

```diff
@@ -56,27 +63,42 @@ termux_step_start_build() {
 				echo "Downloading dependency $PKG@$DEP_VERSION if necessary..."
 			fi
 
-			if [ -e "/data/data/.built-packages/$PKG" ]; then
-				if [ "$(cat "/data/data/.built-packages/$PKG")" = "$DEP_VERSION" ]; then
-					continue
+			if [ -n "$TERMUX_ON_DEVICE_BUILD" ]; then
+				if [ -e "$TERMUX_TOPDIR/.built-packages/$PKG" ]; then
+					if [ "$(cat "$TERMUX_TOPDIR/.built-packages/$PKG")" = "$DEP_VERSION" ]; then
+						continue
+					fi
+				fi
+			else
+				if [ -e "/data/data/.built-packages/$PKG" ]; then
+					if [ "$(cat "/data/data/.built-packages/$PKG")" = "$DEP_VERSION" ]; then
+						continue
+					fi
 				fi
 			fi
 
 			if ! termux_download_deb $PKG $DEP_ARCH $DEP_VERSION; then
 				echo "Download of $PKG@$DEP_VERSION from $TERMUX_REPO_URL failed, building instead"
-				TERMUX_BUILD_IGNORE_LOCK=true ./build-package.sh -a $TERMUX_ARCH -I "${PKG_DIR}"
+				TERMUX_BUILD_IGNORE_LOCK=true ./build-package.sh -I "${PKG_DIR}"
 				continue
 			else
-				if [ ! "$TERMUX_QUIET_BUILD" = true ]; then echo "extracting $PKG..."; fi
-				(
-					cd $TERMUX_COMMON_CACHEDIR-$DEP_ARCH
-					ar x ${PKG}_${DEP_VERSION}_${DEP_ARCH}.deb data.tar.xz
-					tar -xf data.tar.xz --no-overwrite-dir -C /
-				)
+				if [ -z "$TERMUX_ON_DEVICE_BUILD" ]; then
+					if [ ! "$TERMUX_QUIET_BUILD" = true ]; then echo "extracting $PKG..."; fi
+					(
+						cd $TERMUX_COMMON_CACHEDIR-$DEP_ARCH
+						ar x ${PKG}_${DEP_VERSION}_${DEP_ARCH}.deb data.tar.xz
+						tar -xf data.tar.xz --no-overwrite-dir -C /
+					)
+				fi
 			fi
 
-			mkdir -p /data/data/.built-packages
-			echo "$DEP_VERSION" > "/data/data/.built-packages/$PKG"
+			if [ -n "$TERMUX_ON_DEVICE_BUILD" ]; then
+				mkdir -p $TERMUX_TOPDIR/.built-packages
+				echo "$DEP_VERSION" > "$TERMUX_TOPDIR/.built-packages/$PKG"
```

Comment:
Same here, use the new built-packages variable.

Reply:
Thanks.